### PR TITLE
Update GitHub templates

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to:

* Update dependencies listing in existing vizes.
* Add templates for Specviz2D.

Out of scope here because it needs upstream fix:

* #1605 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [x] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature--specviz2d.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+name: 'Feature request: Specviz2D'
+about: Suggest an idea to improve Specviz2D
+title: "[FEAT] Specviz2D:"
+labels: feature, specviz2d
+assignees: ''
+
+---
+
+<!-- This comments are hidden when you submit the issue,
```
Comment (kecnry) [inline .github/ISSUE_TEMPLATE/feature--specviz2d.md:10]: would now be a good excuse to fix this across all issue templates?
```suggestion
<!-- These comments are hidden when you submit the issue,
```

**Review by kecnry — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/feature--specviz2d.md (hunk 1 above)
Comment (pllim) [reply .github/ISSUE_TEMPLATE/feature--specviz2d.md]: Sure. Done. Good catch!

**Comment by pllim** [comment]
What are these dev errors? Look unrelated to this PR though.

https://github.com/spacetelescope/jdaviz/runs/8077753520?check_suite_focus=true

**Comment by pllim** [comment]
I dunno if the dev failure is transient or not. The failures makes no sense. I'll merge this and open new issue if it comes back consistently.

**Comment by pllim** [comment]
Thanks for the reviews!

**Pull request merged**